pkg/common/oras/authprovider/azure/azureworkloadidentity.go

```diff
@@ -71,7 +72,10 @@ func (s *AzureWIProviderFactory) Create(authProviderConfig provider.AuthProvider
 
 	clientID := os.Getenv("AZURE_CLIENT_ID")
 	if clientID == "" {
-		return nil, fmt.Errorf("azure client id environment variable is empty")
+		clientID = conf.ClientID
+		if clientID == "" {
```

Comment:
Looks like the `AZURE_CLIENT_ID` dominates the user-provided `clientID`, just wonder if it's expected.